cmd/soroban-cli/src/commands/contract/bindings/typescript.rs

```diff
@@ -84,34 +87,46 @@ impl NetworkRunnable for Cmd {
     ) -> Result<(), Error> {
         let print = Print::new(global_args.is_some_and(|a| a.quiet));
 
-        let network = self.network.get(&self.locator).ok().unwrap_or_else(|| {
-            network::DEFAULTS
-                .get("testnet")
-                .expect("no network specified and testnet network not found")
-                .into()
-        });
-
-        let contract_id = self
-            .locator
-            .resolve_contract_id(&self.contract_id, &network.network_passphrase)?
-            .0;
-        let contract_address = ScAddress::Contract(Hash(contract_id));
-
-        let spec = if let Some(wasm) = &self.wasm {
+        let (spec, contract_address, rpc_url, network_passphrase) = if let Some(wasm) = &self.wasm {
             print.infoln("Loading contract spec from file...");
             let wasm: wasm::Args = wasm.into();
-            wasm.parse()?.spec
+            (wasm.parse()?.spec, None, None, None)
         } else {
+            let contract_id = self
+                .contract_id
+                .as_ref()
+                .ok_or(Error::MissingWasmOrContractId)?;
+
+            let network = self.network.get(&self.locator).ok().unwrap_or_else(|| {
```

Comment:
I don't think we assume testnet if no network is provided anywhere else, IMO we should align this behavior across commands (so for now we shouldn't assume testnet, and if we do in the future -- we do it for all commands)

Reply:
I agree with that fully, but this is not new behavior. We can address in a separate PR.